client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx

```diff
@@ -499,39 +506,63 @@ export const ApplicationsTable: React.FC = () => {
   };
 
   const discardAssessmentRow = (row: Application) => {
-    setApplicationAssessmentToDiscard(row);
+    setApplicationAssessmentOrReviewToDiscard(row);
     setIsDiscardAssessmentConfirmDialogOpen(true);
   };
 
-  const discardAssessment = (application: Application) => {
-    Promise.all([
-      application.review ? deleteReview(application.review.id!) : undefined,
-    ])
-      .then(() => {
-        const assessment = getApplicationAssessment(application.id!);
-        return Promise.all([
-          assessment ? deleteAssessment(assessment.id!) : undefined,
-        ]);
-      })
-      .then(() => {
+  const {
+    mutate: deleteReview,
+    isError: isDeleteReviewError,
+    error: deleteReviewError,
+  } = useDeleteReviewMutation();
+
+  const {
+    mutate: deleteAssessment,
+    isError: isDeleteAssessmentError,
+    error: deleteAssessmentError,
+  } = useDeleteAssessmentMutation();
+
+  const discardAssessmentAndReview = (application: Application) => {
+    if (application.review?.id) {
+      deleteReview(application.review.id!);
+      if (isDeleteReviewError) {
```

Comment:
This code needs to be handled in an async way. Preferably a callback like we have done elsewhere with react-query.